packages/kbn-apm-synthtrace-client/src/lib/otel/index.ts

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { Fields } from '../entity';
+import { Serializable } from '../serializable';
+import { OtelError } from './error';
+import { OtelMetric } from './metric';
+import { OtelTransaction } from './transaction';
+
+interface OtelSharedResourceAttributes {
+  'service.name'?: string;
+  'agent.name'?: string;
+  'agent.version'?: string;
+  'metricset.interval'?: string;
+  'metricset.name'?: string;
+  'service.instance.id'?: string;
+  'telemetry.sdk.language'?: string;
+  'telemetry.sdk.name'?: string;
+  'telemetry.sdk.version'?: string;
+  'some.resource.attribute'?: string;
+  'timestamp.us'?: number;
```

Comment:
Why is this here under resource attributes? 🤔 This should be only a top level field from what I know.

Reply:
Update: no, I was wrong about this only being top level. `timestamp.us` is indeed not top level.

But it's not under resource attributes , but under attributes:

<img width="500" alt="Screenshot 2024-10-09 at 17 07 40" src="https://github.com/user-attachments/assets/0a4c2590-28e8-42e0-82f9-e6f5c21f7f65">

Reply:
Removed from `resources.attributes` type